src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -507,11 +507,10 @@ def _choose_auto_accelerator(self) -> str:
 
     @staticmethod
     def _choose_gpu_accelerator_backend() -> str:
-        if CUDAAccelerator.is_available():
-            return "cuda"
-
         if MPSAccelerator.is_available():
             return "mps"
+        if CUDAAccelerator.is_available():
+            return "cuda"
```

Comment:
noob q:

what's the reason for changing the order?

Reply:
requested by @carmocca [here](https://github.com/Lightning-AI/lightning/pull/13642#discussion_r926964132)